openpype/hosts/aftereffects/plugins/create/create_render.py

```diff
@@ -1,61 +1,118 @@
+class RenderCreator(Creator):
+    identifier = "render"
+    label = "Render"
+    family = "render"
+    description = "Render creator"
 
-        Creates subsets in format 'familyTaskSubsetname',
-        eg 'renderCompositingMain'.
+    create_allow_context_change = False
 
-        Create only single instance from composition at a time.
-    """
+    def get_icon(self):
+        return resources.get_openpype_splash_filepath()
 
-    name = "renderDefault"
-    label = "Render on Farm"
-    family = "render"
-    defaults = ["Main"]
+    def collect_instances(self):
+        for instance_data in api.list_instances():
+            # legacy instances have family=='render' or 'renderLocal', use them
+            creator_id = (instance_data.get("creator_identifier") or
+                          instance_data.get("family", '').replace("Local", ''))
+            if creator_id == self.identifier:
+                instance_data = self._handle_legacy(instance_data)
+                instance = CreatedInstance.from_existing(
+                    instance_data, self
+                )
+                self._add_instance_to_context(instance)
+
+    def update_instances(self, update_list):
+        created_inst, changes = update_list[0]
+        api.get_stub().imprint(created_inst.get("instance_id"),
+                               created_inst.data_to_store())
+
+    def remove_instances(self, instances):
+        for instance in instances:
+            api.remove_instance(instance)
+            self._remove_instance_from_context(instance)
 
-    def process(self):
-        stub = get_stub()  # only after After Effects is up
-        if (self.options or {}).get("useSelection"):
+    def create(self, subset_name, data, pre_create_data):
+        stub = api.get_stub()  # only after After Effects is up
+        if pre_create_data.get("use_selection"):
             items = stub.get_selected_items(
                 comps=True, folders=False, footages=False
             )
+        else:
+            items = stub.get_items(comps=True, folders=False, footages=False)
+
         if len(items) > 1:
             raise CreatorError(
                 "Please select only single composition at time."
             )
-
         if not items:
             raise CreatorError((
                 "Nothing to create. Select composition "
                 "if 'useSelection' or create at least "
                 "one composition."
             ))
 
-        existing_subsets = [
-            instance['subset'].lower()
-            for instance in list_instances()
+        for inst in self.create_context.instances:
+            if subset_name == inst.subset_name:
+                raise CreatorError("{} already exists".format(
+                    inst.subset_name))
+
+        data["members"] = [items[0].id]
+        new_instance = CreatedInstance(self.family, subset_name, data, self)
+        new_instance.creator_attributes["farm"] = pre_create_data["farm"]
```

Comment:
Pre create data may not be passed (e.g. when called from headless).
```suggestion
        if "farm" in pre_create_data:
            use_farm = pre_create_data["farm"]
            new_instance.creator_attributes["farm"] = use_farm
```